test/profiler/test_profiler.py

```diff
@@ -354,34 +377,41 @@ def trace_handler(p):
             trace_called_num += 1
 
         use_cuda = torch.profiler.ProfilerActivity.CUDA in supported_activities()
-        # Create a temp file to save execution trace data.
+        # Create a temp file to save execution trace and kineto data.
         fp = tempfile.NamedTemporaryFile('w+t', suffix='.et.json', delete=False)
         fp.close()
-        expected_loop_events = 0
-        et = ExecutionTraceObserver()
-        et.register_callback(fp.name)
+        kt = tempfile.NamedTemporaryFile(mode="w+t", suffix=".kineto.json", delete=False)
+        kt.close()
+
+        et = ExecutionTraceObserver().register_callback(fp.name)
         with profile(
             activities=supported_activities(),
             schedule=torch.profiler.schedule(
                 skip_first=3,
                 wait=1,
                 warmup=1,
-                active=2),
+                active=2,
+                repeat=1),
             on_trace_ready=trace_handler,
+            execution_trace_observer=et,
         ) as p:
-            et.start()
             for idx in range(10):
-                expected_loop_events += 1
                 with record_function(f"## LOOP {idx} ##"):
                     self.payload(use_cuda=use_cuda)
                 p.step()
-            et.stop()
 
-        assert trace_called_num == 2
-        assert fp.name == et.get_output_file_path()
+        p.export_chrome_trace(kt.name)
+
+        # Uncomment for debugging
+        # print("Output kineto = ", kt.name)
+        # print("Output ET = ", et.get_output_file_path())
+
+        self.assertEqual(trace_called_num, 1)
+        self.assertEqual(fp.name, et.get_output_file_path())
 
         # cleanup
         et.unregister_callback()
```

Comment:
We can unregister inside the context manager's exit/stop method (https://github.com/pytorch/pytorch/blob/main/torch/profiler/profiler.py#L635) with something like this in there:

```
if self.et: 
  et.unregister_callback()
```

thus allowing users to unregister automatically if they invoke ET from the context manager like we do here.

Reply:
done!